modules/frontend/combiner/common.go

```diff
@@ -129,15 +129,25 @@ func (c *genericCombiner[T]) HTTPFinal() (*http.Response, error) {
 		return nil, err
 	}
 
-	bodyString, err := new(jsonpb.Marshaler).MarshalToString(final)
-	if err != nil {
-		return nil, fmt.Errorf("error marshalling response body: %w", err)
+	var bodyString string
+	if c.marshallingFormat == api.HeaderAcceptProtobuf {
+		buff, err := proto.Marshal(final)
+		if err != nil {
+			return nil, fmt.Errorf("error marshalling response body: %w", err)
+		}
+		bodyString = string(buff)
+	} else {
+		c.marshallingFormat = api.HeaderAcceptJSON
```

Comment:
i find it a little strange to set this here. this feels like something that should be set on creation of the struct to control the response behavior and never changed.

i don't think it currently is causing a bug, but since its unexpected i could see how it would trip someone up.

Reply:
Yes, I know what it looks like. The reason for that is to avoid a scenario where the implementation of the combiner function does not set it and therefore we will return an empty header. Another reason is how difficult is to know what kind of combiner are we returning since we always use the ``Combiner`` interface to later on do a type assertion:

```
NewSearchTagValues(limitBytes).(GRPCCombiner[*tempopb.SearchTagValuesResponse])
```

I wonder if we could return directly 

```
func NewSearchTags(limitBytes int) GRPCCombiner[*tempopb.SearchTagsResponse]
```

Nevertheless I have renamed it to httpMarshalingFormat to be more explicit